teamcity: failed tests on master: TestDockerCLI/test_sql_mem_monitor.tcl

The following tests appear to have failed:

[#82371](https://teamcity.cockroachdb.com/viewLog.html?buildId=82371):

```
--- FAIL: acceptance (proposer evaluated kv)/TestDockerCLI/test_client_side_checking.tcl (146.477s)
null

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/cli/interactive_tests/test_sql_mem_monitor.tcl
+++ b/pkg/cli/interactive_tests/test_sql_mem_monitor.tcl
@@ -72,6 +72,7 @@ set spawn_id $shell_spawn_id
 expect {
     "out of memory" {}
     "cannot allocate memory" {}
+    "std::bad_alloc" {}
     timeout {exit 1}
 }
 eexpect ":/# "

